### ISSUE TITLE
Not able to login with default rekono/rekono credentials

### ISSUE THREAD

**Issue by srikr** [pablosnt/rekono#40]
**Describe the bug**
Not able to login with default rekono/rekono credentials
**To Reproduce**
1. Installed the rekono tool by following below two step:
     ```
     docker-compose build
     docker-compose up -d```
2. In a web browser, Using https://<machine-ip> URL Tried logging in with rekono/rekono credentials, it failed saying Invalid credentials
3. Tried to use` Forgot your password?` but that also did not send email to the given email address.

**Expected behavior**
a) Login should be allowed using rekono/rekono credentials
b) It should be easily allowed to reset the password or username using Forgot your password option

**Screenshots**
![image](https://user-images.githubusercontent.com/49506666/190893275-b2c0c26c-08dd-404e-9196-49ca38bc0cc6.png)

**Environment**
 - OS: Debian 5.18.5-1kali5
 - Browser : Firefox
 - Version : 91.13.0esr (64-bit)

**Installation method**
- [X] Docker - Used this Method
- [ ] CLI
- [ ] From Source

**Additional context**
Add any other context about the problem here.

**Comment by pablosnt** [comment]
Hi @srikr, this bug has been reported by many users. However, the default credentials should work fine in Docker environments. Review this guidelines to get more information about the error, and maybe fix it: https://github.com/pablosnt/rekono/discussions/37#discussioncomment-3614894. Please keep me informed about your progress.

About the password reset, note that default email is rekono@rekono.com and it doesn't exist, so password reset is not available for the default user.

**Comment by srikr** [comment]
Hi @pablosnt , Thanks for the response. I looked in the comment, I am not sure which of the below containers I have to look into for initialize logs:
![image](https://user-images.githubusercontent.com/49506666/190899550-7c251374-d36e-4c2b-9525-da1381315a1a.png)

Can you please point me to the container name and Logs directory.

The API response using the browser development options is 
`500 Internal Server Error`
![image](https://user-images.githubusercontent.com/49506666/190899632-f4d4ba9f-253d-43e0-8613-1ed2b7a76105.png)

**Comment by pablosnt** [comment]
Okay @srikr, you can't see the initialize container because it is executed only once to initialize the database and create the default user, and finally the execution terminates. As the response is a HTTP 500, it's not a credentials problem because the error would be HTTP 401. Probably the problem is because the nginx container is not forwarding correctly the requests to the backend container. Can you try to reboot the backend and the nginx containers in that order please?

I guess it's a puntual error because I use the Docker environment very often and it works fine for me. However, as it's a very common bug, I will review it carefully.

**Comment by srikr** [comment]
Hi @pablosnt, I rebooted first by getting into backend container and then rebooted the nginx container in the same order you mentioned. Then I tried to login, still I see the same 500 Internal Error.

**Comment by srikr** [comment]
Any update @pablosnt ?

**Comment by pablosnt** [comment]
Yes, sorry. I was trying to reproduce the issue. I create a workflow to check the default credentials in Docker environments, and finally I get the same error. I think that the error occurs due to the docker-compose version, because in my local machines I have a recent one:

<img width="451" alt="Captura de Pantalla 2022-09-20 a las 10 43 01" src="https://user-images.githubusercontent.com/69458381/191211576-ea334510-edfc-4f17-80e5-7c01d8a98a34.png">

And in the workflow where the error occurs I have the version 1.29.2: https://github.com/pablosnt/rekono/actions/runs/3088518948/jobs/4995087093#step:3:4210

After review the `docker-compose logs`, I have seen that with the old version, the dependencies between the Docker containers aren't applied, so the credentials don't work because the backend is created before the database initialization.

So, please, can you tell me what docker-compose version do you have? And if it is old, can you upgrade your docker-compose version and retry? Thank you in advance!

**Comment by srikr** [comment]
Sure @pablosnt  You are correct, its exactly the same version:
└```
─# docker-compose version
docker-compose version 1.29.2, build unknown
docker-py version: 5.0.3
CPython version: 3.10.5
OpenSSL version: OpenSSL 3.0.4 21 Jun 2022
```
I will upgrade the docker compose and update this thread.

**Comment by pablosnt** [comment]
It seems that after upgrade docker-compose it works fine: https://github.com/pablosnt/rekono/discussions/37#discussioncomment-3689412. Any update @srikr ?

**Comment by srikr** [comment]
Below are the steps I followed that resulted in same Issue:
1) `curl -L "https://github.com/docker/compose/releases/download/v2.11.1/docker-compose-$(uname -s)-$(uname -m)" -o /usr/local/bin/docker-compose`
2)```
 # docker-compose version
Docker Compose version v2.11.1
```
3) `docker-compose down`
4) ```
docker-compose build
    docker-compose up -d --scale executions-worker=5
```
5)  using firefox browser tried to access https://<vm-ip>/
6) Login Page Came but Login is still failing with same 500 Internal Error

**Comment by pablosnt** [comment]
This should work. Please, remove the local images and run `docker-compose build --no-cache` to ensure you aren't using the old images

**Comment by pablosnt** [comment]
Can you tell me the order of the container logs? I mean, the initialize logs should appear before the backend logs.

About the telegram-bot, you need to pass the Telegram token in the environment variable [`RKN_TELEGRAM_TOKEN`](https://github.com/pablosnt/rekono/wiki/5.-Configuration) to make it works

**Comment by srikr** [comment]
@pablosnt I did rebuild complete things with no-cache, again same 500 Error again. Below is the Snippet of Logs from beginning:

```
initialize  | Traceback (most recent call last):                                                                                                                [0/1841]
initialize  |   File "/code/manage.py", line 22, in <module>
initialize  |     main()
initialize  |   File "/code/manage.py", line 18, in main
initialize  |     execute_from_command_line(sys.argv)
initialize  |   File "/usr/local/lib/python3.9/site-packages/django/core/management/__init__.py", line 419, in execute_from_command_line
initialize  |     utility.execute()
initialize  |   File "/usr/local/lib/python3.9/site-packages/django/core/management/__init__.py", line 413, in execute
initialize  |     self.fetch_command(subcommand).run_from_argv(self.argv)
initialize  |   File "/usr/local/lib/python3.9/site-packages/django/core/management/base.py", line 354, in run_from_argv
initialize  |     self.execute(*args, **cmd_options)
initialize  |   File "/usr/local/lib/python3.9/site-packages/django/core/management/base.py", line 398, in execute
initialize  |     output = self.handle(*args, **options)
initialize  |   File "/code/rekono/management/commands/frontend.py", line 15, in handle
initialize  |     CONFIG.load_config_in_frontend(                                         # Load configuration in frontend .env
initialize  |   File "/code/rekono/config.py", line 96, in load_config_in_frontend
initialize  |     with open(frontend_env_filepath, 'a') as frontend_env:              # Open .env file
initialize  | PermissionError: [Errno 13] Permission denied: '/code/frontend/.env'
redis       | 1:C 20 Sep 2022 14:59:32.529 # oO0OoO0OoO0Oo Redis is starting oO0OoO0OoO0Oo
redis       | 1:C 20 Sep 2022 14:59:32.529 # Redis version=6.2.6, bits=64, commit=00000000, modified=0, pid=1, just started
redis       | 1:C 20 Sep 2022 14:59:32.529 # Warning: no config file specified, using the default config. In order to specify a config file use redis-server /path/to/re
dis.conf
redis       | 1:M 20 Sep 2022 14:59:32.530 * monotonic clock: POSIX clock_gettime
redis       | 1:M 20 Sep 2022 14:59:32.531 * Running mode=standalone, port=6379.
redis       | 1:M 20 Sep 2022 14:59:32.531 # Server initialized
redis       | 1:M 20 Sep 2022 14:59:32.531 # WARNING overcommit_memory is set to 0! Background save may fail under low memory condition. To fix this issue add 'vm.overc
ommit_memory = 1' to /etc/sysctl.conf and then reboot or run the command 'sysctl vm.overcommit_memory=1' for this to take effect.
redis       | 1:M 20 Sep 2022 14:59:32.531 * Ready to accept connections
rekono-executions-worker-5  | 2022-09-20 16:59:37,648 [INFO] - 1 worker -  -  - Worker rq:worker:a29a1305d4e749e4be746144b55aae51: started, version 1.11.0
rekono-executions-worker-5  | 2022-09-20 16:59:37,649 [INFO] - 1 worker -  -  - Subscribing to channel rq:pubsub:a29a1305d4e749e4be746144b55aae51
rekono-executions-worker-5  | 2022-09-20 16:59:37,652 [INFO] - 1 worker -  -  - *** Listening on executions-queue...
rekono-executions-worker-5  | 2022-09-20 16:59:37,656 [INFO] - 1 worker -  -  - Cleaning registries for queue: executions-queue
frontend                    |
frontend                    | > rekono@1.0.0 build
frontend                    | > vue-cli-service build
frontend                    |
frontend                    | Browserslist: caniuse-lite is outdated. Please run:
frontend                    |   npx browserslist@latest --update-db
frontend                    |   Why you should do it regularly: https://github.com/browserslist/browserslist#browsers-data-updating
```

**Comment by pablosnt** [comment]
Can you send me the full docker-compose logs please? If you prefer, you can send it to rekono.project@gmail.com. Only to check that there is nothing strange. Thank you!

**Comment by srikr** [comment]
Attached is complete Logs, Please let me know if you need any other logs
[docker_compose_logs.txt](https://github.com/pablosnt/rekono/files/9609343/docker_compose_logs.txt)

Thanks

**Comment by pablosnt** [comment]
Okey, the initialize container is executed before launch the backend container. It should work. So, can you follow the following steps?

- Remove the Docker containers, images and volumes
- Re-build the Docker environment using `--no-cache` option and re-launch it. **Please, do this from a non-root directory to avoid potential permission errors**

I'm sorry for the generic response, but as I said, it should work and I think that in this case the dependencies between the Docker containers are respected, so maybe the reason could the usage of old images or containers in your system.

Please, keep me informed about your progress with this problem

**Comment by srikr** [comment]
Below is the steps I following based on above inputs:
Step1:
```
└─# who am i
root     pts/6        Aug 31 11:33 (10.102.64.7)
```
Step2:
```
└─# docker volume ls
DRIVER    VOLUME NAME
```
Step3:
```
└─# docker images
REPOSITORY                                     TAG       IMAGE ID       CREATED        SIZE
```
Step4:
```
└─# docker ps
CONTAINER ID   IMAGE                                                 COMMAND                  CREATED       STATUS                  PORTS     NAMES
```
Step5:
`docker-compose build --no-cache`

Step6:
`docker-compose up -d --scale executions-worker=5`
Full Docker Compose Logs: 
[docker_compose_logs.txt](https://github.com/pablosnt/rekono/files/9616505/docker_compose_logs.txt)

Step7:
using firefox browser tried to access https://myvmip/

Step8:
Login Page Came but Login is still failing with same 500 Internal Error

**Comment by srikr** [comment]
Below are Errors seen during initialize:
```
initialize             | Traceback (most recent call last):
initialize              |   File "/code/manage.py", line 22, in <module>
initialize              |     main()
initialize              |   File "/code/manage.py", line 18, in main
initialize              |     execute_from_command_line(sys.argv)
initialize              |   File "/usr/local/lib/python3.9/site-packages/django/core/management/__init__.py", line 419, in execute_from_command_line
initialize              |     utility.execute()
initialize              |   File "/usr/local/lib/python3.9/site-packages/django/core/management/__init__.py", line 363, in execute
initialize              |     settings.INSTALLED_APPS
initialize              |   File "/usr/local/lib/python3.9/site-packages/django/conf/__init__.py", line 82, in __getattr__
initialize              |     self._setup(name)
initialize              |   File "/usr/local/lib/python3.9/site-packages/django/conf/__init__.py", line 69, in _setup
initialize              |     self._wrapped = Settings(settings_module)
initialize                  |   File "/usr/local/lib/python3.9/site-packages/django/conf/__init__.py", line 170, in __init__
initialize                  |     mod = importlib.import_module(self.SETTINGS_MODULE)
initialize                  |   File "/usr/local/lib/python3.9/importlib/__init__.py", line 127, in import_module
initialize                  |     return _bootstrap._gcd_import(name[level:], package, level)
initialize                  |   File "<frozen importlib._bootstrap>", line 1030, in _gcd_import
initialize                  |   File "<frozen importlib._bootstrap>", line 1007, in _find_and_load
initialize                  |   File "<frozen importlib._bootstrap>", line 986, in _find_and_load_unlocked
initialize                  |   File "<frozen importlib._bootstrap>", line 680, in _load_unlocked
initialize                  |   File "<frozen importlib._bootstrap_external>", line 850, in exec_module
initialize                  |   File "<frozen importlib._bootstrap>", line 228, in _call_with_frames_removed
initialize                  |   File "/code/rekono/settings.py", line 65, in <module>
initialize                  |     os.mkdir(dir)
initialize                  | FileExistsError: [Errno 17] File exists: '/rekono/wordlists'
```
Please let me know if you want me to change some line of code in the container directly and do a quick test...

**Comment by pablosnt** [comment]
I think that the reason of these error is that you are using the root user to deploy the Docker environment. I reproduce a similar error in MacOS with root user:
<img width="682" alt="Captura de Pantalla 2022-09-21 a las 15 52 34" src="https://user-images.githubusercontent.com/69458381/191523473-9ff582a5-0eba-4a51-baee-397b93a492b3.png">
Please, retry all the steps with a non-root user and tell me if it works. Thank you in advance!

**Comment by pablosnt** [comment]
It's also important to clone the repository with a non-root user

**Comment by srikr** [comment]
@pablosnt  I did clone rekono using a non root user and started following two commands:
```
docker-compose build
docker-compose up -d --scale executions-worker=5
```
After the end of second command I hit below issue
```
[+] Running 12/14
 ⠿ Container initialize                  Started                                                                                                                   0.4s
 ⠿ Container postfix                     Running                                                                                                                   0.0s
 ⠿ Container redis                       Running                                                                                                                   0.0s
 ⠿ Container telegram-bot                Running                                                                                                                   0.0s
 ⠿ Container postgres                    Started                                                                                                                   0.0s
 ⠿ Container rekono-emails-worker-1      Started                                                                                                                   0.0s
 ⠿ Container frontend                    Waiting                                                                                                                   0.5s
 ⠿ Container rekono-tasks-worker-1       Started                                                                                                                   0.0s
 ⠿ Container rekono-executions-worker-1  Started                                                                                                                   0.0s
 ⠿ Container rekono-executions-worker-4  Started                                                                                                                   0.0s
 ⠿ Container rekono-executions-worker-2  Started                                                                                                                   0.0s
 ⠿ Container rekono-executions-worker-5  Started                                                                                                                   0.0s
 ⠿ Container backend                     Waiting                                                                                                                   0.5s
 ⠿ Container rekono-executions-worker-3  Started                                                                                                                   0.0s
container for service "backend" is unhealthy
```

Now If I rerun the above second command again I am hitting the below errors:
```
└─$ docker-compose up -d --scale executions-worker=5
[+] Running 13/15
 ⠿ Container postfix                     Running                                                                                                                   0.0s
 ⠿ Container redis                       Running                                                                                                                   0.0s
 ⠿ Container initialize                  Started                                                                                                                   0.3s
 ⠿ Container postgres                    Started                                                                                                                   0.0s
 ⠿ Container backend                     Waiting                                                                                                                   0.5s
 ⠿ Container frontend                    Waiting                                                                                                                   0.5s
 ⠿ Container rekono-findings-worker-1    Started                                                                                                                   0.0s
 ⠿ Container telegram-bot                Started                                                                                                                   0.0s
 ⠿ Container rekono-emails-worker-1      Started                                                                                                                   0.0s
 ⠿ Container rekono-executions-worker-1  Started                                                                                                                   0.0s
 ⠿ Container rekono-executions-worker-4  Started                                                                                                                   0.0s
 ⠿ Container rekono-executions-worker-2  Started                                                                                                                   0.0s
 ⠿ Container rekono-executions-worker-3  Started                                                                                                                   0.0s
 ⠿ Container rekono-executions-worker-5  Started                                                                                                                   0.0s
 ⠿ Container rekono-tasks-worker-1       Started                                                                                                                   0.0s
container for service "frontend" is unhealthy
```
Attached is the docker compose logs FYI
[docker_logs.txt](https://github.com/pablosnt/rekono/files/9617813/docker_logs.txt)

@pablosnt In a non root user things are failing long before. I am not sure why we should not run rekono using root user..

**Comment by pablosnt** [comment]
I guess that the problem here is the space available in your disk:

`OSError: [Errno 28] No space left on device: '/rekono/reports'`

Maybe some garbage after build the images and create the volumes many times

**Comment by srikr** [comment]
@pablosnt  Disk got 100% full which is where things were failing with non root user... I cleaned up the disk and did above steps, I am able to login now... Any reason why the same is not working with root user...
I will run the scan and see how much of security issue this tool is finding against our product and get back to you if I have any questions
Thanks and Appreciate your quick responses.

**Comment by pablosnt** [comment]
Amazing! Thank you for keep trying. Honestly, I don't know why it isn't working with root users because I always use a non-root user. But I guess that it's a permission question when some files are copied from your local machine to the Docker container. I can review it deeper for next releases. Thank you again and enjoy Rekono ;)

**Comment by pablosnt** [comment]
Just to let you know. You can use Docker environment using `root` user, but first, you need to provide write access to all system users in all Rekono source code. The reason is that Rekono folder is shared with the Docker containers, some files are written from there and Docker is probably executed with a regular user. This files are things like wordlists, frontend `.env` file, execution outputs, etc. My advice is still using Rekono in Docker environments with regular users instead of using privileged ones.

**Comment by pablosnt** [comment]
The error in Docker environment using `root` user has been fixed in the [new release 1.2.0](https://github.com/pablosnt/rekono/releases/tag/1.2.0)
